packages/arb-avm-cpp/avm/include/avm/machinestate/machinestate.hpp

```diff
@@ -103,34 +103,44 @@ struct MachineOutput {
 };
 
 struct MachineStateKeys {
```

Comment:
I think the changes to `MachineStateKeys` would be somewhat simpler by just adding the three new fields (`l1_block_number`, `l2_block_number`, and `last_inbox_timestamp`) to `MachineOutput` instead. That change will propagate to affect some of the refactoring and I think provides a cleaner split since then a checkpoint without a machine is just a `MachineOutput`

Reply:
:+1: